core/primitives/src/congestion_info.rs

```diff
@@ -280,8 +290,10 @@ impl CongestionInfo {
     pub fn remove_delayed_receipt_gas(&mut self, gas: Gas) -> Result<(), RuntimeError> {
         match self {
             CongestionInfo::V1(inner) => {
-                inner.delayed_receipts_gas =
-                    inner.delayed_receipts_gas.checked_sub(gas as u128).ok_or_else(|| {
+                inner.delayed_receipts_gas = inner
+                    .delayed_receipts_gas
+                    .checked_sub(gas.as_gas() as u128)
```

Comment:
I believe this should work (there is at least one more place below in this file that we can change `u64 as u128` with explicit and safe `.into()`):

```suggestion
                    .checked_sub(gas.as_gas().into())
```